master/internal/stream/system.go

```diff
@@ -199,21 +191,26 @@ func (ps *PublisherSet) Websocket(
 		bootemChan = ps.bootemChan
 	}()
 
+	// First read the startup message.
+	var startupMsg StartupMsg
+	err := socket.ReadJSON(&startupMsg)
+	// XXX: (how) do we want the server to communicate errors to the streaming clients
+	// check if error was due to something other than the connection closing.
+	if err != nil && !websocket.IsCloseError(err, websocket.CloseAbnormalClosure) {
+		log.Errorf("unable to read startup message: %s", err)
+	}
+	if err != nil {
+		return nil
+	}
```

Comment:
nit: have one `err != nil` statement, and use a separate log statement inside of it to log the exceptional error.

That better expresses what's going on: we're _gonna_ stop if there's an error, and within that case, there's another case where we might log something.